src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -111,10 +111,12 @@ static List<Type> GetHandlerMessageTypes(List<Type> list)
                     {
                         var messageType = @interface.GetGenericArguments()[0];
                         foundMessageTypes.Add(messageType);
+
+                        foundMessageTypes.AddRange(messageType.Assembly.GetTypes().Where(t=> messageType.IsAssignableFrom(t)));
```

Comment:
Stackoverflow doesn't come up with anything better :( http://stackoverflow.com/questions/1268397/how-to-find-all-the-types-in-an-assembly-that-inherit-from-a-specific-type-c-sha

And yea, if that type is in yet another assembly we're screwed again